### PR TITLE
Improve DistSQL parser test engine with db-types for issue#11014

### LINKED ISSUE #11014: Improve DistSQL parser test engine with db-types
The current DistSQL parser test engine using `db-types` to recognize DistSQL or normal SQL, which is not appropriate.

Please refactor here, decouple `db-types` and DistSQL.

### PR DESCRIPTION
**Pull request by totalo**
Fixes #11014 .

Changes proposed in this pull request:
- Improve DistSQL parser test engine with db-types

### CODE REVIEW

**Comment by totalo** [comment]
@terrymanu hi，I have finished the first version, please review it.

File: shardingsphere-test/shardingsphere-parser-test/src/test/java/org/apache/shardingsphere/test/sql/parser/parameterized/DistSQLParserParameterizedTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.SQLParserTestCase;
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.distsql.loader.DistSQLCasesLoader;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class DistSQLParserParameterizedTest {
```
Comment (tristaZero) [inline shardingsphere-test/shardingsphere-parser-test/src/test/java/org/apache/shardingsphere/test/sql/parser/parameterized/DistSQLParserParameterizedTest.java:36]: Well, this PR looks nearly perfect, but please give a look at whether it is better to be a final class and remove the redundant blank line 64.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-parser-test/src/test/java/org/apache/shardingsphere/test/sql/parser/parameterized/DistSQLParserParameterizedTest.java (hunk 1 above)
Comment (totalo) [reply shardingsphere-test/shardingsphere-parser-test/src/test/java/org/apache/shardingsphere/test/sql/parser/parameterized/DistSQLParserParameterizedTest.java:36]: Yes, it should be fianl.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/rdl/create.xml [hunk 2]
```diff
@@ -17,14 +17,14 @@
...
-    <sql-case id="create-static-readwrite-splitting-rule" value="CREATE READWRITE_SPLITTING RULE ms_group_0 (WRITE_RESOURCE=primary_ds, READ_RESOURCES(replica_ds_0,replica_ds_1), TYPE(NAME=random)))" db-types="DistSQL" />
-    <sql-case id="create-dynamic-readwrite-splitting-rule" value="CREATE READWRITE_SPLITTING RULE ms_group_1(AUTO_AWARE_RESOURCE=group_0, TYPE(NAME=random,PROPERTIES(read_weight='2:1')))" db-types="DistSQL" />
-    <sql-case id="create-database-discovery-rule" value="CREATE DB_DISCOVERY RULE ha_group_0 (RESOURCES(resource0,resource1), TYPE(NAME=mgr,PROPERTIES(groupName='92504d5b-6dec',keepAliveCron=''))), ha_group_1 (RESOURCES(resource2,resource3), TYPE(NAME=mgr2,PROPERTIES(groupName='92504d5b-6dec-2',keepAliveCron='')))" db-types="DistSQL" />
-    <sql-case id="create-encrypt-rule" value="CREATE ENCRYPT RULE t_encrypt (RESOURCE=ds_1, COLUMNS((NAME=user_id,PLAIN=user_plain,CIPHER=user_cipher,TYPE(NAME=AES,PROPERTIES('aes-key-value'='123456abc'))), (NAME=order_id, CIPHER =order_cipher,TYPE(NAME=MD5))))" db-types="DistSQL" />
+    <distSQL-case id="add-resource-single-without-password" value="ADD RESOURCE ds_0(HOST=127.0.0.1,PORT=3306,DB=test0,USER=ROOT);" />
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/rdl/create.xml:20]: How about change the xml tag `distSQL-case` to `distsql-case`?

**Review by terrymanu — APPROVED** [review]

**Review by tristaZero — APPROVED** [review]

**Pull request merged**